pages/form.tsx

```diff
@@ -1,66 +1,201 @@
 import { fetchHasura } from "../src/lib/hasura";
 import { Input } from "@codegouvfr/react-dsfr/Input";
+import { RadioButtons } from "@codegouvfr/react-dsfr/RadioButtons";
+import { Checkbox } from "@codegouvfr/react-dsfr/Checkbox";
+
+import { generateFormData } from "../src/services/fake-form-data";
 
 import {
+  EncryptedFormLocalState,
+  base64UrlDecode,
+  encryptFile,
   encryptFormData,
   initializeEncryptedFormLocalState,
 } from "@socialgouv/e2esdk-crypto";
 import { insert_one } from "../src/queries/form";
 import Button from "@codegouvfr/react-dsfr/Button";
 import Alert from "@codegouvfr/react-dsfr/Alert";
+import { z } from "zod";
+import { fr } from "@codegouvfr/react-dsfr";
+import { useIsDark } from "@codegouvfr/react-dsfr/useIsDark";
+
+const formPublicKeyString = "AatyS2mCJd__zewF-mT_IEd4925CQgf-CC9U3U3ZRnk"; // form public key
+const nameFingerprint = "73ZLcYHHAhx6rgMbgpPgc0F1qE89oWOAq6UGdrHkkJQ"; // workspace nameFingerprint
 
 type PostVariables = {
   submissionBucketId: string;
   signature: string;
   sealedSecret: string;
   publicKey: string;
+  answersFiles: any[];
   data: string;
 };
 
+export const jsonDataSchema = z.object({
+  firstName: z.string(),
+  lastName: z.string(),
+  message: z.string(),
+  email: z.string().email().optional(),
+  color: z.string().optional(),
+  newsletter: z.boolean().optional().default(false),
+  alerts: z.boolean().optional().default(false),
+  files: z.array(z.any()),
+});
+
+type FormData = z.infer<typeof jsonDataSchema>;
+
+function generateSubmissions() {
+  // generate and submit bunch of fake submissions
+  const rows = Array.from({ length: 100 }, () => generateFormData());
+  console.time("encryptAndSubmitForm");
+  return Promise.all(rows.map(encryptAndSubmitForm)).then(() => {
+    console.timeEnd("encryptAndSubmitForm");
+  });
+}
+
+const encryptAndSubmitForm = async (data: Record<string, any>) => {
+  const formPublicKey = base64UrlDecode(formPublicKeyString);
+  const state = await initializeEncryptedFormLocalState(
+    nameFingerprint, //submissionBucketId,
+    formPublicKey
+  );
+
+  const answersFiles = [];
+  if (data.files) {
+    const { files } = data;
+    delete data.files;
+    const encryptedFiles = await Promise.all(
+      files.map((file: File) => readAndEncryptFile(file, state))
+    );
+    // console.log("encryptedFiles", encryptedFiles);
+    const formData = new FormData();
+    for (let i = 0; i < encryptedFiles.length; i++) {
+      const { encryptedFile, metadata } = encryptedFiles[i];
+      console.dir({ encryptedFile, metadata }, { depth: Infinity });
+      formData.set(`file_${i}`, encryptedFile);
+    }
+    if (Array.from(formData.values()).flat().length > 0) {
+      await fetch("/api/upload-answers-files", {
+        method: "POST",
+        body: formData,
+      });
+      for (const { metadata } of encryptedFiles) {
+        const { hash, key } = metadata;
+        answersFiles.push({ file_hash: hash, file_key: key });
```

Comment:
**note**: Sending the key like this to the server would allow it to decrypt files, so it's not e2ee. The key is encrypted as part of the `encryptFormData` call, so to keep track of things you'd only need to memorise the file hashes.

Reply:
OK that's what it seemed to me, so, if I understand well we have to put key in form data that will be encrypted, thanks

Reply:
Yes, if you look at [the fullstack example](https://github.com/SocialGouv/e2esdk/blob/59aa11a9f369d6cc3ce0f5985a5e62559a2f8d58/examples/fullstack/contact-forms/src/lib/contact-form.tsx#L116), you'll see the metadata (which contains the file name, size and also the decryption key) is passed to the form data to encrypt.